packages/pharos/src/components/checkbox/pharos-checkbox.ts

```diff
@@ -64,15 +64,25 @@ export class PharosCheckbox extends FormMixin(FormElement) {
   }
 
   public onChange(): void {
+    const originalIndeterminateState = this.indeterminate;
     this.indeterminate = false;
     this.checked = this._checkbox.checked;
 
-    this.dispatchEvent(
-      new Event('change', {
+    const notCancelled = this.dispatchEvent(
+      new CustomEvent('change', {
         bubbles: true,
+        cancelable: true,
         composed: true,
+        detail: {
+          target: this._checkbox,
+        },
       })
     );
+
+    if (!notCancelled) {
+      this.indeterminate = originalIndeterminateState;
+      this.checked = !this._checkbox.checked;
```

Comment:
Wonder if we want to update this one to use the same "original state reset" form as the indeterminate, for consistency

Reply:
is this what you meant?
https://github.com/ithaka/pharos/pull/431/files#diff-18ffa349736aa11f0cc47dfdcb4e2129f167d12b920d4e7e82ead64a6734a524R67-R68

i do think that reads better